### ISSUE TITLE
PHP7 support

### ISSUE THREAD

**Issue by muxx** [tony2001/pinba_extension#23]
Hi, Anton! Do you plan to make php7 version of pinba extension?

**Comment by tony2001** [comment]
I do, but it's not an immediate priority atm.

**Comment by tony2001** [comment]
PHP7 version is available in php7 branch of this repository.
Would be great if you give it a try and report all the issues you find.

**Comment by dimsua** [comment]
php 7.0.0RC3
/bin/sh /admin/dims/phpbuild/7.0/pinba_extension/libtool --mode=compile cc -DNDEBUG -I. -I/admin/dims/phpbuild/7.0/pinba_extension -DPHP_ATOM_INC -I/admin/dims/phpbuild/7.0/pinba_extension/include -I/admin/dims/phpbuild/7.0/pinba_extension/main -I/admin/dims/phpbuild/7.0/pinba_extension -I/usr/local/php70/include/php -I/usr/local/php70/include/php/main -I/usr/local/php70/include/php/TSRM -I/usr/local/php70/include/php/Zend -I/usr/local/php70/include/php/ext -I/usr/local/php70/include/php/ext/date/lib  -DHAVE_CONFIG_H  -m64 -O3   -c /admin/dims/phpbuild/7.0/pinba_extension/pinba-pb-c.c -o pinba-pb-c.lo
mkdir .libs
 cc -DNDEBUG -I. -I/admin/dims/phpbuild/7.0/pinba_extension -DPHP_ATOM_INC -I/admin/dims/phpbuild/7.0/pinba_extension/include -I/admin/dims/phpbuild/7.0/pinba_extension/main -I/admin/dims/phpbuild/7.0/pinba_extension -I/usr/local/php70/include/php -I/usr/local/php70/include/php/main -I/usr/local/php70/include/php/TSRM -I/usr/local/php70/include/php/Zend -I/usr/local/php70/include/php/ext -I/usr/local/php70/include/php/ext/date/lib -DHAVE_CONFIG_H -m64 -O3 -c /admin/dims/phpbuild/7.0/pinba_extension/pinba-pb-c.c  -fPIC -DPIC -o .libs/pinba-pb-c.o
/bin/sh /admin/dims/phpbuild/7.0/pinba_extension/libtool --mode=compile cc -DNDEBUG -I. -I/admin/dims/phpbuild/7.0/pinba_extension -DPHP_ATOM_INC -I/admin/dims/phpbuild/7.0/pinba_extension/include -I/admin/dims/phpbuild/7.0/pinba_extension/main -I/admin/dims/phpbuild/7.0/pinba_extension -I/usr/local/php70/include/php -I/usr/local/php70/include/php/main -I/usr/local/php70/include/php/TSRM -I/usr/local/php70/include/php/Zend -I/usr/local/php70/include/php/ext -I/usr/local/php70/include/php/ext/date/lib  -DHAVE_CONFIG_H  -m64 -O3   -c /admin/dims/phpbuild/7.0/pinba_extension/pinba.c -o pinba.lo
 cc -DNDEBUG -I. -I/admin/dims/phpbuild/7.0/pinba_extension -DPHP_ATOM_INC -I/admin/dims/phpbuild/7.0/pinba_extension/include -I/admin/dims/phpbuild/7.0/pinba_extension/main -I/admin/dims/phpbuild/7.0/pinba_extension -I/usr/local/php70/include/php -I/usr/local/php70/include/php/main -I/usr/local/php70/include/php/TSRM -I/usr/local/php70/include/php/Zend -I/usr/local/php70/include/php/ext -I/usr/local/php70/include/php/ext/date/lib -DHAVE_CONFIG_H -m64 -O3 -c /admin/dims/phpbuild/7.0/pinba_extension/pinba.c  -fPIC -DPIC -o .libs/pinba.o
/admin/dims/phpbuild/7.0/pinba_extension/pinba.c: In function ‘php_pinba_key_compare’:
/admin/dims/phpbuild/7.0/pinba_extension/pinba.c:997: error: too many arguments to function ‘string_compare_function’

**Comment by tony2001** [comment]
I just pushed a fix for that.

**Comment by dimsua** [comment]
now compiling fine and send data.

**Comment by nacc** [comment]
Is there a plan to cut this into a release at some point? Ubuntu is going to be PHP7.0 only with 16.04 and we either need to drop our package altogether, or (ideally) package a released version, rather than a git snapshot.

**Comment by muxx** [comment]
The php7 version of extension works fine! Thanks!

**Comment by linko22** [comment]
Cannot build branch php7 under centos 6, php 7.0.5

Configuring for:
PHP Api Version:         20151012
Zend Module Api No:      20151012
Zend Extension Api No:   320151012

libtool: compile:  cc -DNDEBUG -I. -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -DPHP_ATOM_INC -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/include -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/main -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -c /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba-pb-c.c  -fPIC -DPIC -o .libs/pinba-pb-c.o
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:369: error: expected ')' before '*' token
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:384:90: error: macro "zend_hash_index_update" passed 5 arguments, but takes just 3
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:393: error: expected ')' before '*' token
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'php_pinba_dict_find_or_add':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:490: warning: passing argument 2 of 'zend_hash_find' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:154: note: expected 'struct zend_string *' but argument is of type 'char *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:490: error: too many arguments to function 'zend_hash_find'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:493:68: error: macro "zend_hash_add" passed 6 arguments, but takes just 3
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:493: error: 'zend_hash_add' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:493: error: (Each undeclared identifier is reported only once
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:493: error: for each function it appears in.)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function '_pinba_fetch_global_var':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:507: error: used struct type value where scalar is required
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:508: error: invalid type argument of 'unary *' (have 'zval')
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:508: error: invalid type argument of 'unary *' (have 'zval')
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:508: error: invalid type argument of 'unary *' (have 'zval')
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:508: error: invalid type argument of 'unary *' (have 'zval')
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:508: warning: passing argument 2 of 'zend_hash_find' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:154: note: expected 'struct zend_string *' but argument is of type 'char *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:508: error: too many arguments to function 'zend_hash_find'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:508: warning: comparison between pointer and integer
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:509: warning: implicit declaration of function 'Z_TYPE_PP'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:509: warning: implicit declaration of function 'Z_STRLEN_PP'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:511: warning: implicit declaration of function 'Z_STRVAL_PP'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:511: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:511: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:511: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:511: warning: passing argument 1 of 'strlen' makes pointer from integer without a cast
/usr/include/string.h:399: note: expected 'const char *' but argument is of type 'int'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:511: warning: passing argument 2 of 'memcpy' makes pointer from integer without a cast
/usr/include/bits/string3.h:49: note: expected 'const void * __restrict__' but argument is of type 'int'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:511: warning: passing argument 1 of '__strdup' makes pointer from integer without a cast
/usr/include/bits/string2.h:1303: note: expected 'const char *' but argument is of type 'int'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:659:70: error: macro "zend_hash_sort" passed 4 arguments, but takes just 3
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'php_create_pinba_packet':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:659: error: 'zend_hash_sort' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:665: warning: passing argument 2 of 'zend_hash_get_current_data_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:171: note: expected 'HashPosition *' but argument is of type 'void **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:665: error: too many arguments to function 'zend_hash_get_current_data_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:679: warning: passing argument 2 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'struct zend_string **' but argument is of type 'char **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:679: warning: passing argument 3 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'zend_ulong *' but argument is of type 'uint *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:679: warning: passing argument 4 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'HashPosition *' but argument is of type 'ulong *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:679: error: too many arguments to function 'zend_hash_get_current_key_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:701: warning: passing argument 2 of 'zend_hash_get_current_data_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:171: note: expected 'HashPosition *' but argument is of type 'void **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:701: error: too many arguments to function 'zend_hash_get_current_data_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:714: warning: passing argument 2 of 'zend_hash_find' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:154: note: expected 'struct zend_string *' but argument is of type 'char *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:714: error: too many arguments to function 'zend_hash_find'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:719:102: error: macro "zend_hash_add" passed 6 arguments, but takes just 3
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:719: error: 'zend_hash_add' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:726: warning: passing argument 2 of 'zend_hash_get_current_data_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:171: note: expected 'HashPosition *' but argument is of type 'void **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:726: error: too many arguments to function 'zend_hash_get_current_data_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:757: warning: passing argument 2 of 'zend_hash_get_current_data_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:171: note: expected 'HashPosition *' but argument is of type 'void **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:757: error: too many arguments to function 'zend_hash_get_current_data_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:763: warning: passing argument 2 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'struct zend_string **' but argument is of type 'char **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:763: warning: passing argument 3 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'zend_ulong *' but argument is of type 'uint *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:763: warning: passing argument 4 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'HashPosition *' but argument is of type 'ulong *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:763: error: too many arguments to function 'zend_hash_get_current_key_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:808: warning: passing argument 2 of 'zend_hash_get_current_data_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:171: note: expected 'HashPosition *' but argument is of type 'void **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:808: error: too many arguments to function 'zend_hash_get_current_data_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'php_pinba_flush_data':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:924: error: 'php_pinba_timer_stop_helper' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'php_pinba_key_compare':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:979: error: 'Bucket' has no member named 'nKeyLength'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:980: error: lvalue required as left operand of assignment
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:983: error: lvalue required as left operand of assignment
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:984: error: 'Bucket' has no member named 'arKey'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:985: error: 'Bucket' has no member named 'nKeyLength'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:988: error: 'Bucket' has no member named 'nKeyLength'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:989: error: lvalue required as left operand of assignment
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:992: error: lvalue required as left operand of assignment
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:993: error: 'Bucket' has no member named 'arKey'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:994: error: 'Bucket' has no member named 'nKeyLength'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:997: error: too many arguments to function 'string_compare_function'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1037:70: error: macro "zend_hash_sort" passed 4 arguments, but takes just 3
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'php_pinba_array_to_tags':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1037: error: 'zend_hash_sort' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1041:54: error: macro "zend_hash_get_current_data" passed 2 arguments, but takes just 1
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1041: error: 'zend_hash_get_current_data' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1047: error: 'IS_BOOL' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1050: warning: initialization from incompatible pointer type
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1051: warning: passing argument 1 of 'zval_get_type' from incompatible pointer type
/usr/include/php/Zend/zend_types.h:326: note: expected 'const struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1051: warning: passing argument 1 of 'zval_get_type' from incompatible pointer type
/usr/include/php/Zend/zend_types.h:326: note: expected 'const struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1051: warning: passing argument 1 of '_convert_to_string' from incompatible pointer type
/usr/include/php/Zend/zend_operators.h:243: note: expected 'struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1052: warning: passing argument 1 of '_estrndup' makes pointer from integer without a cast
/usr/include/php/Zend/zend_alloc.h:85: note: expected 'const char *' but argument is of type 'int'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1062: warning: passing argument 2 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'struct zend_string **' but argument is of type 'char **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1062: warning: passing argument 3 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'zend_ulong *' but argument is of type 'uint *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1062: warning: passing argument 4 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'HashPosition *' but argument is of type 'ulong *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1062: error: too many arguments to function 'zend_hash_get_current_key_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'php_pinba_get_timer_info':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1117: warning: implicit declaration of function 'MAKE_STD_ZVAL'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1122: error: too many arguments to function 'add_assoc_stringl_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1129: warning: passing argument 1 of 'zval_add_ref' from incompatible pointer type
/usr/include/php/Zend/zend_variables.h:141: note: expected 'struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'pinba_client_free_storage':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1154: warning: implicit declaration of function 'STR_FREE'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: At top level:
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1165: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'pinba_client_new'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_timer_start':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1237: warning: implicit declaration of function 'INIT_PZVAL'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1244: warning: assignment makes integer from pointer without a cast
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1254: warning: implicit declaration of function 'zend_list_addref'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1255: warning: implicit declaration of function 'RETURN_RESOURCE'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_timer_add':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1312: warning: assignment makes integer from pointer without a cast
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_timer_stop':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1339: warning: implicit declaration of function 'ZEND_FETCH_RESOURCE'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1339: error: expected expression before 'pinba_timer_t'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_timer_delete':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1362: error: expected expression before 'pinba_timer_t'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1369: warning: passing argument 1 of 'zend_list_delete' makes pointer from integer without a cast
/usr/include/php/Zend/zend_list.h:59: note: expected 'struct zend_resource *' but argument is of type 'int'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_timer_data_merge':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1390: error: expected expression before 'pinba_timer_t'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1398:128: error: macro "zend_hash_merge" passed 6 arguments, but takes just 4
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1398: error: 'zend_hash_merge' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1378: warning: unused variable 'tmp'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_timer_data_replace':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1421: error: expected expression before 'pinba_timer_t'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1426: warning: passing argument 1 of '_zval_ptr_dtor' from incompatible pointer type
/usr/include/php/Zend/zend_variables.h:112: note: expected 'struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1431: warning: passing argument 1 of '_zval_ptr_dtor' from incompatible pointer type
/usr/include/php/Zend/zend_variables.h:112: note: expected 'struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_timer_tags_merge':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1464: error: expected expression before 'pinba_timer_t'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_timer_tags_replace':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1529: error: expected expression before 'pinba_timer_t'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_flush':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1563: warning: passing argument 1 of 'zval_get_type' from incompatible pointer type
/usr/include/php/Zend/zend_types.h:326: note: expected 'const struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1563: warning: passing argument 1 of 'zval_get_type' from incompatible pointer type
/usr/include/php/Zend/zend_types.h:326: note: expected 'const struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1563: warning: passing argument 1 of '_convert_to_string' from incompatible pointer type
/usr/include/php/Zend/zend_operators.h:243: note: expected 'struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1564: warning: passing argument 1 of 'php_pinba_flush_data' makes pointer from integer without a cast
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:920: note: expected 'const char *' but argument is of type 'int'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_get_info':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1581: error: 'zend_rsrc_list_entry' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1581: error: 'le' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1625: error: too many arguments to function 'add_assoc_string_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1627: error: too many arguments to function 'add_assoc_string_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1631: error: too many arguments to function 'add_assoc_string_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1633: error: too many arguments to function 'add_assoc_string_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1637: error: too many arguments to function 'add_assoc_string_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1639: error: too many arguments to function 'add_assoc_string_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1642:66: error: macro "add_assoc_string" passed 4 arguments, but takes just 3
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1642: error: 'add_assoc_string' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1648: error: too many arguments to function 'zend_hash_get_current_data_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1668: warning: passing argument 2 of 'zend_hash_get_current_data_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:171: note: expected 'HashPosition *' but argument is of type 'void **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1668: error: too many arguments to function 'zend_hash_get_current_data_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1674: warning: passing argument 2 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'struct zend_string **' but argument is of type 'char **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1674: warning: passing argument 3 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'zend_ulong *' but argument is of type 'uint *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1674: warning: passing argument 4 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'HashPosition *' but argument is of type 'ulong *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1674: error: too many arguments to function 'zend_hash_get_current_key_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1675:45: error: macro "add_assoc_string" passed 4 arguments, but takes just 3
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1703:34: error: macro "RETVAL_STRINGL" passed 3 arguments, but takes just 2
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_get_data':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1703: error: 'RETVAL_STRINGL' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_timer_get_info':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1720: error: expected expression before 'pinba_timer_t'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_timers_stop':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1730: error: 'zend_rsrc_list_entry' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1730: error: 'le' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1739: error: too many arguments to function 'zend_hash_get_current_data_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_timers_get':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1754: error: 'zend_rsrc_list_entry' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1754: error: 'le' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1765: error: too many arguments to function 'zend_hash_get_current_data_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1774: warning: passing argument 2 of 'add_next_index_resource' makes pointer from integer without a cast
/usr/include/php/Zend/zend_API.h:429: note: expected 'struct zend_resource *' but argument is of type 'int'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1904:81: error: macro "zend_hash_update" passed 6 arguments, but takes just 3
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_tag_set':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1904: error: 'zend_hash_update' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_tag_get':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1920: warning: passing argument 2 of 'zend_hash_find' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:154: note: expected 'struct zend_string *' but argument is of type 'char *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1920: error: too many arguments to function 'zend_hash_find'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1920: warning: comparison between pointer and integer
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1923:25: error: macro "RETURN_STRING" passed 2 arguments, but takes just 1
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1923: error: 'RETURN_STRING' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_tag_delete':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1938: warning: passing argument 2 of 'zend_hash_del' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:146: note: expected 'struct zend_string *' but argument is of type 'char *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1938: error: too many arguments to function 'zend_hash_del'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zif_pinba_tags_get':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1958: warning: passing argument 2 of 'zend_hash_get_current_data_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:171: note: expected 'HashPosition *' but argument is of type 'void **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1958: error: too many arguments to function 'zend_hash_get_current_data_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1964: warning: passing argument 2 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'struct zend_string **' but argument is of type 'char **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1964: warning: passing argument 3 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'zend_ulong *' but argument is of type 'uint *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1964: warning: passing argument 4 of 'zend_hash_get_current_key_ex' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:168: note: expected 'HashPosition *' but argument is of type 'ulong *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1964: error: too many arguments to function 'zend_hash_get_current_key_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1965:49: error: macro "add_assoc_string" passed 4 arguments, but takes just 3
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1965: error: 'add_assoc_string' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient___construct':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1987: warning: implicit declaration of function 'zend_object_store_get_object'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1987: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1990:53: error: macro "zend_hash_get_current_data" passed 2 arguments, but takes just 1
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1990: error: 'zend_hash_get_current_data' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1994: warning: passing argument 1 of 'zval_get_type' from incompatible pointer type
/usr/include/php/Zend/zend_types.h:326: note: expected 'const struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1994: warning: passing argument 1 of 'zval_get_type' from incompatible pointer type
/usr/include/php/Zend/zend_types.h:326: note: expected 'const struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1994: warning: passing argument 1 of '_convert_to_string' from incompatible pointer type
/usr/include/php/Zend/zend_operators.h:243: note: expected 'struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:1996: warning: passing argument 1 of 'php_pinba_parse_server' makes pointer from integer without a cast
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:197: note: expected 'char *' but argument is of type 'int'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_setHostname':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2030: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_setServername':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2031: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_setScriptname':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2032: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_setSchema':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2033: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_setRequestCount':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2054: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_setMemoryFootprint':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2055: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_setMemoryPeak':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2056: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_setDocumentSize':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2057: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_setStatus':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2058: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_setRequestTime':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2059: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_setRusage':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2073: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2081:52: error: macro "zend_hash_get_current_data" passed 2 arguments, but takes just 1
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2081: error: 'zend_hash_get_current_data' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2084: warning: passing argument 1 of 'zval_get_type' from incompatible pointer type
/usr/include/php/Zend/zend_types.h:326: note: expected 'const struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2084: warning: passing argument 1 of 'convert_to_double' from incompatible pointer type
/usr/include/php/Zend/zend_operators.h:245: note: expected 'struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2085: warning: implicit declaration of function 'Z_DVAL_PP'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_setTag':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2102: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2106:89: error: macro "zend_hash_update" passed 6 arguments, but takes just 3
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2106: error: 'zend_hash_update' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'php_pinba_client_timer_add_set':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2126: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2151:53: error: macro "zend_hash_get_current_data" passed 2 arguments, but takes just 1
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2151: error: 'zend_hash_get_current_data' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2154: warning: passing argument 1 of 'zval_get_type' from incompatible pointer type
/usr/include/php/Zend/zend_types.h:326: note: expected 'const struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2154: warning: passing argument 1 of 'convert_to_double' from incompatible pointer type
/usr/include/php/Zend/zend_operators.h:245: note: expected 'struct zval *' but argument is of type 'struct zval **'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2184: warning: passing argument 2 of 'zend_hash_find' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:154: note: expected 'struct zend_string *' but argument is of type 'char *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2184: error: too many arguments to function 'zend_hash_find'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2197:115: error: macro "zend_hash_add" passed 6 arguments, but takes just 3
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2197: error: 'zend_hash_add' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2200:117: error: macro "zend_hash_update" passed 6 arguments, but takes just 3
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2200: error: 'zend_hash_update' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_send':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2233: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zim_PinbaClient_getData':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2261: warning: cast to pointer from integer of different size
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2269:34: error: macro "RETVAL_STRINGL" passed 3 arguments, but takes just 2
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2269: error: 'RETVAL_STRINGL' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'OnUpdateCollectorAddress':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2477: warning: passing argument 1 of 'strlen' from incompatible pointer type
/usr/include/string.h:399: note: expected 'const char *' but argument is of type 'struct zend_string *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2477: warning: passing argument 1 of '__strdup' from incompatible pointer type
/usr/include/bits/string2.h:1303: note: expected 'const char *' but argument is of type 'struct zend_string *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2504: error: 'new_value_length' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2504: warning: passing argument 6 of 'OnUpdateString' makes integer from pointer without a cast
/usr/include/php/Zend/zend_ini.h:162: note: expected 'int' but argument is of type 'void *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2504: error: too many arguments to function 'OnUpdateString'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zm_startup_pinba':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2538: error: 'php_timer_resource_dtor' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2546: warning: assignment from incompatible pointer type
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2547: warning: assignment from incompatible pointer type
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2551: error: too many arguments to function 'zend_register_internal_class_ex'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2552: error: 'pinba_client_new' undeclared (first use in this function)
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'zm_activate_pinba':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2596: warning: passing argument 3 of '_zend_hash_init' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:57: note: expected 'dtor_func_t' but argument is of type 'void (*)(void *)'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2597: warning: passing argument 3 of '_zend_hash_init' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:57: note: expected 'dtor_func_t' but argument is of type 'void (*)(void *)'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2609: warning: passing argument 1 of 'zend_is_auto_global' from incompatible pointer type
/usr/include/php/Zend/zend_compile.h:775: note: expected 'struct zend_string *' but argument is of type 'char *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2609: error: too many arguments to function 'zend_is_auto_global'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2612: error: used struct type value where scalar is required
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2612: error: invalid type argument of 'unary *' (have 'zval')
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2612: error: invalid type argument of 'unary *' (have 'zval')
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2612: error: invalid type argument of 'unary *' (have 'zval')
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2612: error: invalid type argument of 'unary *' (have 'zval')
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2612: warning: passing argument 2 of 'zend_hash_find' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:154: note: expected 'struct zend_string *' but argument is of type 'char *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2612: error: too many arguments to function 'zend_hash_find'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2612: warning: comparison between pointer and integer
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2613: warning: passing argument 1 of '_estrndup' makes pointer from integer without a cast
/usr/include/php/Zend/zend_alloc.h:85: note: expected 'const char *' but argument is of type 'int'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2616: error: used struct type value where scalar is required
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2616: error: invalid type argument of 'unary *' (have 'zval')
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2616: error: invalid type argument of 'unary *' (have 'zval')
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2616: error: invalid type argument of 'unary *' (have 'zval')
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2616: error: invalid type argument of 'unary *' (have 'zval')
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2616: warning: passing argument 2 of 'zend_hash_find' from incompatible pointer type
/usr/include/php/Zend/zend_hash.h:154: note: expected 'struct zend_string *' but argument is of type 'char *'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2616: error: too many arguments to function 'zend_hash_find'
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2616: warning: comparison between pointer and integer
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:2617: warning: passing argument 1 of '_estrndup' makes pointer from integer without a cast
/usr/include/php/Zend/zend_alloc.h:85: note: expected 'const char *' but argument is of type 'int'
make: *** [pinba.lo] Error 1
make: *** Waiting for unfinished jobs....

**Comment by tony2001** [comment]
That's not php7 branch.

**Comment by linko22** [comment]
[root@linko pinba_extension]# git status
# On branch php7
nothing to commit, working directory clean

[root@linko pinba_extension]# git remote show origin
* remote origin
  Fetch URL: https://github.com/tony2001/pinba_extension.git
  Push  URL: https://github.com/tony2001/pinba_extension.git
  HEAD branch: master
  Remote branches:
    antoxa_master tracked
    autotimers    tracked
    master        tracked
    php7          tracked
    pinba_class   tracked
    protobuf_c    tracked
    request_tags  tracked
  Local branch configured for 'git pull':
    master merges with remote master
  Local refs configured for 'git push':
    master pushes to master (up to date)
    php7   pushes to php7   (local out of date)

So, i take all files from here.

**Comment by tony2001** [comment]
These are clearly different directories:
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/
[root@linko pinba_extension]

**Comment by tony2001** [comment]
https://github.com/tony2001/pinba_extension/archive/php7.zip you can download zipped version here.
But you .spec file apparently still points to the old sources.

**Comment by linko22** [comment]
I double checked. All files and archives are fine, no other.
I will try to use your link.

**Comment by linko22** [comment]
Yes! It compile now! Thanks!
With some non-critical warnings

`/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'php_pinba_init_socket':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:440: warning: 'ai_ptr' may be used uninitialized in this function
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c: In function 'php_create_pinba_packet':
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:543: warning: 'tag_ids' may be used uninitialized in this function
/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c:543: warning: 'tag_value_ids' may be used uninitialized in this function
`

**Comment by tony2001** [comment]
Those warnings are false, but I committed a fix to silence them.
You can download the archive again, they should be gone now.

**Comment by linko22** [comment]
Yep, all clean now.
`+ /usr/bin/make -j4
/bin/sh /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/libtool --mode=compile cc -DNDEBUG -I. -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -DPHP_ATOM_INC -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/include -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/main -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -c /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba-pb-c.c -o pinba-pb-c.lo
/bin/sh /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/libtool --mode=compile cc -DNDEBUG -I. -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -DPHP_ATOM_INC -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/include -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/main -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -c /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c -o pinba.lo
/bin/sh /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/libtool --mode=compile cc -DNDEBUG -I. -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -DPHP_ATOM_INC -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/include -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/main -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -c /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/protobuf-c.c -o protobuf-c.lo
libtool: compile:  cc -DNDEBUG -I. -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -DPHP_ATOM_INC -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/include -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/main -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -c /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba.c  -fPIC -DPIC -o .libs/pinba.o
libtool: compile:  cc -DNDEBUG -I. -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -DPHP_ATOM_INC -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/include -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/main -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -c /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/protobuf-c.c  -fPIC -DPIC -o .libs/protobuf-c.o
libtool: compile:  cc -DNDEBUG -I. -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -DPHP_ATOM_INC -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/include -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/main -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -c /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/pinba-pb-c.c  -fPIC -DPIC -o .libs/pinba-pb-c.o
/bin/sh /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/libtool --mode=link cc -DPHP_ATOM_INC -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/include -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/main -I/home/rpm/rpmbuild/BUILD/php-pinba-1.1.0 -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -o pinba.la -export-dynamic -avoid-version -prefer-pic -module -rpath /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/modules  pinba-pb-c.lo pinba.lo protobuf-c.lo
libtool: link: cc -shared  .libs/pinba-pb-c.o .libs/pinba.o .libs/protobuf-c.o    -m64 -mtune=generic   -Wl,-soname -Wl,pinba.so -o .libs/pinba.so
libtool: link: ( cd ".libs" && rm -f "pinba.la" && ln -s "../pinba.la" "pinba.la" )
/bin/sh /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/libtool --mode=install cp ./pinba.la /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/modules
libtool: install: cp ./.libs/pinba.so /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/modules/pinba.so
libtool: install: cp ./.libs/pinba.lai /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/modules/pinba.la
libtool: finish: PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/sbin" ldconfig -n /home/rpm/rpmbuild/BUILD/php-pinba-1.1.0/modules
`

**Comment by tony2001** [comment]
Thanks!

**Comment by curlup** [comment]
@tony2001 Is there any plans to provide php7 release?

**Comment by dimsua** [comment]
https://github.com/tony2001/pinba_extension/tree/php7 work fine on php 7.0 and 7.1

**Comment by bamsell** [comment]
@dimsua your link is dead :(

**Comment by sewelseb** [comment]
@dimsua , your link looks to be dead :(

**Comment by tony2001** [comment]
That's because PHP7 support is now merged into the master branch and available by default.

**Comment by sewelseb** [comment]
Thank you Tony :)

**Comment by wapmorgan** [comment]
Does this work on 7.3 or 7.4?

**Comment by yuklia** [comment]
@wapmorgan no, it doesn't 
i tried to build on `FROM php:7.3-fpm`  and got en error:
```
/usr/src/php/ext/pinba/pinba.c: In function 'zif_pinba_timer_start':
/usr/src/php/ext/pinba/pinba.c:1295:19: error: lvalue required as increment operand
  GC_REFCOUNT(rsrc)++;
                   ^~
/usr/src/php/ext/pinba/pinba.c: In function 'zif_pinba_timer_add':
/usr/src/php/ext/pinba/pinba.c:1354:19: error: lvalue required as increment operand
  GC_REFCOUNT(rsrc)++;
                   ^~
/usr/src/php/ext/pinba/pinba.c: In function 'zif_pinba_timers_get':
/usr/src/php/ext/pinba/pinba.c:1811:21: error: lvalue required as increment operand
    GC_REFCOUNT(rsrc)++;
                     ^~
make: *** [Makefile:195: pinba.lo] Error 1
```
